src/modules/plus/filter_dynamicgpstext.yml

```diff
@@ -0,0 +1,321 @@
+        * #gps_dist_r#        - the raw total distance so far (if present in gps file)
+        * #gps_dist_c#        - the computed total travelled distance
+        * #gps_datetime_now#  - date and time of current gps point shown (UTC)
+        * #file_datetime_now# - date and time of current frame in video file (absolute)
+        * #hr#                - heart rate (if present in gps file)
+        * #gps_bearing#       - current GPS bearing (degrees 0-360)
+        * #gps_compass#       - current GPS direction (8 divisions: N, NE, E, etc)
+        * #gps_vdist_up#      - total GPS altitude gain so far (computed)
+        * #gps_vdist_down#    - total GPS altitude loss so far
+        * #gps_dist_uphill#   - total distance travelled uphill so far
+        * #gps_dist_downhill# - total distance travelled downhill so far
+        * #gps_dist_flat#     - total distance travelled on flat area so far
+      The r in keyword stands for "raw" and c for "computed". If a keyword can't 
+      produce valid data it returns "--".
+      Time-based keywords can include a strftime format string to customize the
+      output as long as you put some delimiter except "#" between the keyword
+      and the format string and the keyword comes first. For example, 
+      "#localtime %I:%M:%S %p#" shows only the time in 12-hour format.
+      Speed and distance keywords may include an extra format keyword to convert 
+      the value to metric/imperial units. Default is meters and km/h respectively.
+      Supported formats, distance: m|meter, km|kilometer*, mi|mile*, ft|feet;
+      speed: ms|m/s|meter, km|km/h|kilometer, mi|mi/h|mile, ft|ft/s|feet.
+      Computed values are calculated since begining of GPS file or since 
+      "gps_processing_start_time" property, if set.
+      The # may be escaped with "\".
+    required: yes
+    readonly: no
+    default: >
+      Speed: #gps_speed_c#km/h\n
+      Distance: #gps_dist_c#m\n
+      Altitude: #gps_elev#m\n\n
+      GPS time: #gps_datetime_now#\n
+      GPS location: #gps_lat_c#, #gps_lon_c#
+    widget: text
+
+  - identifier: gps.file
+    title: File/URL
+    type: string
+    description: >
+      "The fullpath of file containing location (GPS) data. Supported extensions: .gpx, .tcx."
+    required: no
+    readonly: no
+    widget: fileopen
+    
+  - identifier: major_offset
+    title: Major offset
+    type: integer
+    description: >
+      An offset (in seconds) to be added to the video file to match it to the GPS track.
+      Most of the time this will at least need to be set to the timezone difference 
+      between the 2 files plus some seconds if the video recording device isn't 
+      precisely set to correct time. GPS time is always exact and in UTC. 
+      Use positive values if GPS is ahead of video and negative otherwise.
+    default: 0
+    readonly: no
+    required: no
+    mutable: yes
+    widget: text
+    
+  - identifier: minor_offset
```

Comment:
I find it cumbersome that there are two offsets. Could there just be one offset and let the application be responsible for adding two different values if that is needed?

Reply:
The minor_offset property appeared as a bugfix to the shotcut ui i made (it lost the value it stored in some circumstances, like after closing and opening the project, I think even if minimizing the window). The UI element is extermely useful and I would really not want to remove it, but the filter property I don't care about. 

On the other hand it's completely optional so I didn't think it would matter much.

Reply:
Maybe it will be more obvious to me when I see the filter UI. But based on the YML text, I am imagining the UI has some abstract settings that might be difficult for users to understand. Why can't the UI just have one offset?

Reply:
The major offset has textboxes for days/hours/mins/seconds, the minor one is just a slider with a -60 to 60 seconds range that can be changed (and processed in realtime) using the mouse scrollwheel. Of course I can just move the minor value into the major one but in practice I noticed I almost always end up splitting the video in multiple clips and just keep the major offset identical across the gps file while adjusting the minor one a few seconds forward/backwards to match some obvious movements in the particular clip.

![Screenshot 2021-04-18 filter ui](https://user-images.githubusercontent.com/3215812/115156548-69281700-a08d-11eb-934b-4a8e1bb89a29.png)